runtime/win32.c

```diff
@@ -620,12 +622,22 @@ void caml_win32_unregister_overflow_detection(void)
 
 /* Seeding of pseudo-random number generators */
 
-int caml_win32_random_seed (intnat data[16])
+int caml_win32_random_seed(intnat data[16])
 {
-  /* For better randomness, consider:
-     http://msdn.microsoft.com/library/en-us/seccrypto/security/rtlgenrandom.asp
-     http://blogs.msdn.com/b/michael_howard/archive/2005/01/14/353379.aspx
-  */
+  unsigned char buffer[12];
+
+  /* Try kernel entropy first */
+  NTSTATUS st = BCryptGenRandom(NULL, buffer, sizeof(buffer),
+                                BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+  if (NT_SUCCESS(st)) {
```

Comment:
This is probably equivalent, but `NT_SUCCESS` checks for a range of result codes; wouldn't it be simpler to compare with `STATUS_SUCCESS` here?

Reply:
OK, thanks!